iothub/device/src/Pipeline/DefaultDelegatingHandler.cs

```diff
@@ -178,6 +178,23 @@ protected private virtual void Dispose(bool disposing)
             }
         }
 
+        public virtual Task<DateTime> RefreshTokenAsync(CancellationToken cancellationToken)
+        {
+            ThrowIfDisposed();
+            return NextHandler?.RefreshTokenAsync(cancellationToken) ?? Task.FromResult(DateTime.MaxValue);
+        }
+
+        public virtual DateTime GetRefreshesOn(CancellationToken cancellationToken)
+        {
+            ThrowIfDisposed();
+            return NextHandler?.GetRefreshesOn(cancellationToken) ?? DateTime.MaxValue;
+        }
+
+        public virtual void SetRefreshesOn(CancellationToken cancellationToken)
+        {
+            ThrowIfDisposed();
```

Comment:
Out of curiosity, what is this method meant to do? I'm not too familiar with this part of our code so this isn't a critique, just a question.

Reply:
We are refreshing a SAS token every 1 hour and we want to keep track of when it needs to be refreshed. In order to do that, we need this method to set RefreshesOn variable. We use RefreshesOn variable in RefreshLoopAsync to check if the wait time is met.

Reply:
Which method, `ThrowIfDisposed()`?

If the underlying object is disposed, it will throw `ObjectDisposedException`.